lib/matplotlib/figure.py

```diff
@@ -1595,39 +1435,21 @@ def gca(self, **kwargs):
         %(Axes_kwdoc)s
 
         """
-        ckey, cax = self._axstack.current_key_axes()
-        # if there exists an axes on the stack see if it matches
-        # the desired axes configuration
-        if cax is not None:
-
-            # if no kwargs are given just return the current axes
-            # this is a convenience for gca() on axes such as polar etc.
-            if not kwargs:
-                return cax
-
-            # if the user has specified particular projection detail
-            # then build up a key which can represent this
-            else:
-                projection_class, _, key = \
-                    self._process_projection_requirements(**kwargs)
-
-                # let the returned axes have any gridspec by removing it from
-                # the key
-                ckey = ckey[1:]
-                key = key[1:]
-
-                # if the cax matches this key then return the axes, otherwise
-                # continue and a new axes will be created
-                if key == ckey and isinstance(cax, projection_class):
-                    return cax
-                else:
-                    _api.warn_external('Requested projection is different '
-                                       'from current axis projection, '
-                                       'creating new axis with requested '
-                                       'projection.')
-
-        # no axes found, so create one which spans the figure
-        return self.add_subplot(1, 1, 1, **kwargs)
+        if kwargs:
+            _api.warn_deprecated(
+                "3.4",
+                message="Calling gca() with keyword arguments is deprecated. "
+                "In a future version, gca() will take no keyword arguments. "
+                "The gca() function should only be used to get the current "
+                "axes, or if no axes exist, create new axes with default "
+                "keyword arguments. To create a new axes with non-default "
+                "arguments, use plt.axes() or plt.subplot().")
+        cax = self._axstack()
+        if cax is None:
+            # no axes found, so create one which spans the figure
+            return self.add_subplot(1, 1, 1, **kwargs)
+        else:
+            return cax
```

Comment:
Can be simplified to:
```suggestion
        if self._axstack.empty():
            return self.add_subplot(1, 1, 1, **kwargs)
        else:
            return self._axstack()
```